src/commons/workspace/WorkspaceReducer.ts

```diff
@@ -353,17 +382,35 @@ export const WorkspaceReducer: Reducer<WorkspaceManagerState> = (
       };
     case EVAL_INTERPRETER_SUCCESS:
       const execType = state[workspaceLocation].context.executionMethod;
+      const tokens = state[workspaceLocation].tokenCount;
       const newOutputEntry: Partial<ResultOutput> = {
         type: action.payload.type as 'result' | undefined,
         value: execType === 'interpreter' ? action.payload.value : stringify(action.payload.value)
       };
 
       lastOutput = state[workspaceLocation].output.slice(-1)[0];
       if (lastOutput !== undefined && lastOutput.type === 'running') {
-        newOutput = state[workspaceLocation].output.slice(0, -1).concat({
+        const newOutputEntryWithLogs = {
           consoleLogs: lastOutput.consoleLogs,
           ...newOutputEntry
-        } as ResultOutput);
+        } as ResultOutput;
+        const notificationOutputs: NotificationOutput[] = [];
+        if (state[workspaceLocation].hasTokenCounter) {
+          notificationOutputs.push({
+            consoleLog: `This program has ${tokens} tokens.`,
+            type: 'notification'
+          });
+        }
+        const customNotification = state[workspaceLocation].customNotification;
```

Comment:
Is this for future plans, or already used?

Reply:
This is most likely for future plans. Not yet used.

Reply:
Would it make more sense for `customNotification` to be `customNotifications` (a `string[]`) instead then? Since it's reasonable to assume you can have either 0, 1, or more than 1; there's no reason to only have either 0 or 1 notification present.

Anyway, it's not used now, so I'll just approve the PR, but perhaps you can create an issue: either remove it (cause it's unused), or make it support multiple notifications?